runelite-client/src/main/java/net/runelite/client/plugins/roguesden/RoguesDenOverlay.java

```diff
@@ -67,28 +79,144 @@ public Dimension render(Graphics2D graphics)
 		{
 			if (tile.getPlane() == client.getPlane() && obstacle.getLocalLocation().distanceTo(playerLocation) < MAX_DISTANCE)
 			{
-				Polygon p = tile.getGameObjects()[0].getConvexHull();
-				if (p != null)
+				Area clickBox = obstacle.getClickbox();
+				if (clickBox != null)
+				{
+					Point mouse = client.getMouseCanvasPosition();
+					if (clickBox.contains(mouse.getX(), mouse.getY()))
+					{
+						graphics.setColor(COL_OBJ_BORDER_HOVER);
+					}
+					else
+					{
+						graphics.setColor(COL_OBJ_BORDER);
+					}
+
+					graphics.draw(clickBox);
+					graphics.setColor(COL_OBJ);
+					graphics.fill(clickBox);
+
+				}
+				else
 				{
-					graphics.setColor(Color.CYAN);
-					graphics.drawPolygon(p);
+					Polygon p;
+					if (obstacle instanceof GameObject)
+						p = ((GameObject) obstacle).getConvexHull();
+					else
+						p = obstacle.getCanvasTilePoly();
+
+					if (p != null)
+					{
+						graphics.setColor(COL_OBJ);
+						graphics.drawPolygon(p);
+					}
 				}
 			}
 		});
 
-		plugin.getObstaclesTile().forEach((obstacle, tile) ->
+		for (Obstacles.Obstacle obstacle : Obstacles.OBSTACLES)
 		{
-			if (tile.getPlane() == client.getPlane() && obstacle.getLocalLocation().distanceTo(playerLocation) < MAX_DISTANCE)
+			LocalPoint localPoint = LocalPoint.fromWorld(client, obstacle.tile);
+
+			if (localPoint != null && obstacle.tile.getPlane() == client.getPlane() && localPoint.distanceTo(playerLocation) < MAX_DISTANCE)
 			{
-				Polygon p = obstacle.getCanvasTilePoly();
-				if (p != null)
+				Point mp = Perspective.worldToMiniMap(client, localPoint.getX(), localPoint.getY());
+				if (mp != null)
```

Comment:
Use braces for all if statements, there's a quite a few other occurrences of this as well.